extensions/standard-processors/processors/AttributesToJSON.cpp

```diff
@@ -94,10 +95,11 @@ void AttributesToJSON::onSchedule(core::ProcessContext* context, core::ProcessSe
 }
 
 bool AttributesToJSON::isCoreAttributeToBeFiltered(const std::string& attribute) const {
-  return !include_core_attributes_ && core_attributes_.find(attribute) != core_attributes_.end();
+  const auto& special_attributes = core::SpecialFlowAttribute::getSpecialFlowAttributes();
+  return !include_core_attributes_ && ranges::find(special_attributes, attribute) != ranges::end(special_attributes);
 }
 
-std::unordered_set<std::string> AttributesToJSON::getAttributesToBeWritten(const std::map<std::string, std::string>& flowfile_attributes) const {
+std::unordered_set<std::string> AttributesToJSON::getAttributesToBeWritten(core::FlowFile::AttributeMap* flowfile_attributes) const {
```

Comment:
Having a pointer parameter that we're not checking for null before dereferencing is a recipe for disaster. Can we change it to const ref? We don't seem to modify the AttributeMap.

Reply:
Good point, we can safely change it as the pointer is just the returned address of the internal member. Updated in effc866d914fb3ccff4f70a7a0ebcea6305403fc